src/app/compose/compose.component.ts

```diff
@@ -123,15 +123,21 @@ export class ComposeComponent implements AfterViewInit, OnDestroy, OnInit {
                     this.draftDeskservice.froms[0].nameAndAddress : '';
             } else {
                 this.model.from = from.nameAndAddress;
-                if ( from.is_signature_html ) {
-                    this.model.useHTML = true;
-                }
                 if ( !this.has_pasted_signature && from.signature ) {
                     this.has_pasted_signature = true;
                     this.signature = from.signature;
                     this.model.msg_body = this.signature.concat('\n\n', this.model.msg_body);
+                    if ( from.is_signature_html ) {
+                        this.model.useHTML = true;
+                        this.model.html = this.signature.concat('\n\n', this.model.html || this.model.msg_body);
+                    }
                 }
             }
+            // This.. shouldnt happen as only have content if reply/fwd
```

Comment:
DId it happen anyway in testing? What's the story here?

Reply:
Nope, it just seemed illogical to leave out that check...

Reply:
Fair enough :)